File: Website/src/app/dtos/stat-parameters.ts
import { ParamMap } from '@angular/router';

export class StatParameters {
    team: number;
    era: number;
    season: number;
    league: number;
    playerType: number;
    seasonType: number;
    selectedColumnIndex: number;

    setParams(params: ParamMap) {
        this.team = +params.get('team');
        this.era = +params.get('era');
        this.season = +params.get('season');
        this.playerType = +params.get('plyType');
        this.league = +params.get('league');
        this.seasonType = +params.get('ssnType');
        this.selectedColumnIndex = +params.get('col') || -1;
    }
}

File: Service/Dtos/StatParameters.cs
using BojoBox.EntityFramework.Entities;
using System.Collections.Generic;

namespace BojoBox.Service.Dtos
{
    public class StatParametersDto
    {
        public int? Team { get; set; }
        public int? Era { get; set; }
        public int? Season { get; set; }
        public int? League { get; set; }
        public int? SeasonType { get; set; }
        public int? Col { get; set; }
        public int? Page { get; set; }
    }
}
